Dockerfile

```diff
@@ -0,0 +1,45 @@
+ARG BASE_IMAGE=ubuntu:22.04
+FROM ${BASE_IMAGE} AS build-stage
+
+ARG PJ_TAG=3.9.2
+
+###############################################################################
+# Install general dependencies.
+###############################################################################
+RUN apt update && apt install -y build-essential cmake git 
+
+###############################################################################
+# Compile PlotJuggler
+###############################################################################
+WORKDIR /plotjuggler_ws/src
+RUN git clone --branch ${PJ_TAG} https://github.com/facontidavide/PlotJuggler.git PlotJuggler
+# Install PlotJuggler dependencies.
+RUN apt update && apt -y install qtbase5-dev libqt5svg5-dev libqt5websockets5-dev \
+      libqt5opengl5-dev libqt5x11extras5-dev libprotoc-dev libzmq3-dev \
+      liblz4-dev libzstd-dev
+# Build PlotJuggler. This will take a long time.
+WORKDIR /plotjuggler_ws
+RUN cmake -S src/PlotJuggler -B build/PlotJuggler -DCMAKE_INSTALL_PREFIX=install
+RUN cmake --build build/PlotJuggler --config RelWithDebInfo --target install
+
+###############################################################################
+# Compile the plugin
+###############################################################################
+ARG ADD_UNITS=OFF
+
+COPY . /apbin_plugin
```

Comment:
COPY --link . /apbin_plugin to gain some rebuild time by putting the file into an temp container before copying.